pkg/console/operator/sync_v400.go

```diff
@@ -38,60 +40,114 @@ import (
 // The next loop will pick up where they previous left off and move the process forward one step.
 // This ensures the logic is simpler as we do not have to handle coordination between objects within
 // the loop.
-func sync_v400(co *consoleOperator, operatorConfig *operatorv1.Console, consoleConfig *configv1.Console) (*operatorv1.Console, *configv1.Console, bool, error) {
+func sync_v400(co *consoleOperator, originalOperatorConfig *operatorv1.Console, consoleConfig *configv1.Console) (bool, error) {
+	errors := []error{}
+	operatorConfig := originalOperatorConfig.DeepCopy()
 	logrus.Println("running sync loop 4.0.0")
 	recorder := co.recorder
 
 	// track changes, may trigger ripples & update operator config or console config status
 	toUpdate := false
 
-	rt, rtChanged, rtErr := SyncRoute(co, operatorConfig)
-	if rtErr != nil {
-		return operatorConfig, consoleConfig, toUpdate, rtErr
+	rt, rtChanged, err := SyncRoute(co, operatorConfig)
```

Comment:
I'm thinking this over. It may actually break things.  We've been depending on the sync loop stopping as soon as it hits one error & starting over.  This has been a good thing, as the loop will resolve 1 problem and die. A change will happen, so the loop gets triggered again.  On the next pass, it will resolve the next problem & die. It will do this over and over until all errors are resolved & everything is working.  

If we go the `append` route, then an early error doesn't stop the loop.  That means a later `Sync*` func is probably trying to fix something it may not be able to fix.

For example, if the `route` does not yet have `Spec.Host` (this is provided by the cluster), then we don't really care if it even tries to create the configmap, the data will be wrong.  And we don't really care if it tries to make the deployment, it can't run without the correct configmap.

The `return` for each of these ensures that we only have to deal with the single relevant error.

Reply:
I wonder if we can use a couple funcs for this:

```golang
        // still kill the loop...
	rt, rtChanged, rtErr := SyncRoute(co, operatorConfig)
	if rtErr != nil {
                 // helper func to deal with the status update
		handleSyncErrorCondition(co, operatorConfig, fmt.Errorf("%q: %v", "route", rtErr))
		return operatorConfig, consoleConfig, toUpdate, rtErr
	}
	toUpdate = toUpdate || rtChanged

// function defined below...
func handleSyncErrorCondition(co *consoleOperator, operatorConfig *operatorv1.Console, message string) {
	v1helpers.SetOperatorCondition(&operatorConfig.Status.Conditions, operatorv1.OperatorCondition{
		Type:               workloadFailingCondition,
		Status:             operatorv1.ConditionTrue,
		Message:            message,
		Reason:             "SyncError",
		LastTransitionTime: metav1.Now(),
	})

	if _, err := co.operatorConfigClient.UpdateStatus(operatorConfig); err != nil {
		// handle
	}
	// hanlde
}
```
Still tinkering with this.